### PR TITLE
Adding ontap_cluster_peer.py Modules

### PR DESCRIPTION
**Pull request by carchi8py**
##### SUMMARY
Add ontap_cluster_peer.py Modules

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME
- na_ontap_cluster_peer.py

##### ANSIBLE VERSION

```
bash-3.2# ansible --version
ansible 2.7.0.dev0 (sf_commonfiles caf71e24dd) last updated 2018/08/06 10:46:57 (GMT -700)
  config file = None
  configured module search path = [u'/Users/chrisarchibald/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /private/tmp/ansible/lib/ansible
  executable location = /private/tmp/ansible/bin/ansible
  python version = 2.7.12 (default, Oct 11 2016, 05:20:59) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.38)]
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: lib/ansible/modules/storage/netapp/na_ontap_cluster_peer.py [hunk 1]
```diff
@@ -114,7 +114,7 @@ def __init__(self):
 
         self.module = AnsibleModule(
             argument_spec=self.argument_spec,
-            required_together=['source_intercluster_lif', 'dest_intercluster_lif', 'passphrase'],
+            required_together=(['source_intercluster_lif', 'dest_intercluster_lif', 'passphrase']),
```
Comment (lonico) [inline lib/ansible/modules/storage/netapp/na_ontap_cluster_peer.py]: According to documentation, we should have two sets of [ ] :

required_together
This is self explanatory, all specified parameters need to be specified together:

required_together=[
  [ 'a', 'b' ]
]

**Review by schmots1 — APPROVED** [review]
shipit

File: lib/ansible/modules/storage/netapp/na_ontap_cluster_peer.py [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+    """
+
+    def __init__(self):
+
+        self.argument_spec = netapp_utils.ontap_sf_host_argument_spec()
+        self.argument_spec.update(dict(
+            state=dict(required=False, type='str', choices=['present', 'absent'], default='present'),
+            source_intercluster_lif=dict(required=False, type='str'),
+            dest_intercluster_lif=dict(required=False, type='str'),
+            passphrase=dict(required=False, type='str'),
```
Comment (gundalow) [inline lib/ansible/modules/storage/netapp/na_ontap_cluster_peer.py]: `, no_log=True)`

File: lib/ansible/modules/storage/netapp/na_ontap_cluster_peer.py [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+    - name: Create cluster peer
+      na_ontap_cluster_peer:
+        state: present
+        source_intercluster_lif: 1.2.3.4
+        dest_intercluster_lif: 5.6.7.8
+        passphrase: XXXX
+        hostname: "{{ netapp_hostname }}"
+        username: "{{ netapp_username }}"
+        password: "{{ netapp_password }}"
+        dest_hostname: "{{ netapp_hostname }}"
```
Comment (gundalow) [inline lib/ansible/modules/storage/netapp/na_ontap_cluster_peer.py]: Does this example make sense, are you peering to yourself?

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/storage/netapp/na_ontap_cluster_peer.py (hunk 3 above)
Comment (carchi8py) [reply lib/ansible/modules/storage/netapp/na_ontap_cluster_peer.py]: @gundalow we are peering to a second netapp host. I'll update the documentation to say {{ destination_netapp_hostname }} to clear that up

**Review by gundalow — APPROVED** [review]

**Pull request merged**